ocp_resources/utils.py

```diff
@@ -26,27 +27,93 @@ class TimeoutSampler:
     Yielding the output allows you to handle every value as you wish.
 
     Feel free to set the instance variables.
+
+    exceptions_dict should be in the following format:
+    {
+        exception0: [exception0_msg0],
+        exception1: [
+            exception1_msg0,
+            exception1_msg1
+        ],
+        exception2: []
+    }
+
+    Args:
+        wait_timeout (int): Time in seconds to wait for func to return a value equating to True
+        sleep (int): Time in seconds between calls to func
+        func (function): to be wrapped by TimeoutSampler
+        exceptions (tuple): Deprecated: Tuple containing all retry exceptions to pass to TimeoutSampler
+        exceptions_msg (str): Deprecated: String to match exception against
+        exceptions_dict (dict): Definition for exception handling
+        print_log (bool): Print elapsed time to log
     """
 
     def __init__(
         self,
         wait_timeout,
         sleep,
         func,
-        exceptions=None,
-        exceptions_msg=None,
+        exceptions=None,  # Deprecated
+        exceptions_msg=None,  # Deprecated
+        exceptions_dict=None,
         print_log=True,
         **func_kwargs,
     ):
         self.wait_timeout = wait_timeout
         self.sleep = sleep
         self.func = func
         self.func_kwargs = func_kwargs
-        self.exception = exceptions or Exception
         self.elapsed_time = None
-        self.exceptions_msg = exceptions_msg
         self.print_log = print_log
 
+        self.exceptions_dict = {}
```

Comment:
`self.exceptions_dict = exceptions_dict or {}
`

And please add TODO once deprecated, once _pre_process_exceptions is removed, should be:

self.exceptions_dict = exceptions_dict or {Exception: []}

Reply:
I will add the todo, but it will remain `self.exceptions_dict = {}` until pre-process is removed.